drivers/hmis/app/models/hmis/hud/processors/enrollment_processor.rb

```diff
@@ -39,5 +46,23 @@ def assign_metadata
         data_source_id: @processor.hud_user.data_source_id,
       )
     end
+
+    private def assign_unit(unit_id)
+      return unless unit_id.present?
+
+      enrollment = @processor.send(factory_name)
+      unit = enrollment.project.units.find(unit_id)
+      raise "Unit not found: #{unit_id}" unless unit.present?
+
+      active_unit_occupancy = enrollment.active_unit_occupancy
+      # If already assigned to this unit: do nothing
+      return if active_unit_occupancy&.unit_id == unit.id
```

Comment:
We don't, but we already have to look up the `enrollment.active_unit_occupancy` here in order to unassign the current unit, so I figure we can check here rather than making an extra query to look up the occupants.